Mathlib/Data/ENNReal/Basic.lean

```diff
@@ -510,6 +510,16 @@ theorem toNNReal_nat (n : ℕ) : (n : ℝ≥0∞).toNNReal = n := by
 theorem toReal_nat (n : ℕ) : (n : ℝ≥0∞).toReal = n := by
   rw [← ENNReal.ofReal_natCast n, ENNReal.toReal_ofReal (Nat.cast_nonneg _)]
 
+lemma _root_.Nat.cast_toNNReal (N : ℕ) :
+    (Nat.cast N : ℝ≥0∞).toNNReal = (Nat.cast N : ℝ).toNNReal := by
+  rw [Real.toNNReal_of_nonneg (by positivity), ENNReal.toNNReal_nat]
+  congr
+
+lemma _root_.Nat.rpow_mul_toNNReal (k : ℕ) (R : ℝ) :
+    2 ^ k * R.toNNReal = (2 ^ k * R).toNNReal := by
```

Comment:
This doesn't seem like the statement you want (i.e., it's specialized to `2`). Probably you meant something else?